### PR TITLE
Adds for migration of connection strings to appsettings.json

### PR DESCRIPTION
**Pull request by johanbenschop**
- [x] You've read the [Contributor Guide](../CONTRIBUTING.md) and [Code of Conduct](../CODE-OF-CONDUCT.md).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

**PR Title**
Adds for migration of connection strings to appsettings.json

**PR Description**
Added IUpdater<ConfigFile> implementation to connection strings.
Based on the existing implementation of AppSettingsConfigUpdater.
Includes changes to the integration tests.

Addresses #64

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/upgrade-assistant?pullRequest=571) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/upgrade-assistant?pullRequest=571) before we can accept your contribution.<br/><br/>:x: johanbenschop [sign now](https://cla.dotnetfoundation.org/dotnet/upgrade-assistant?pullRequest=571)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/upgrade-assistant?pullRequest=571) it.</sub>

File: src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/ConnectionStringsConfigUpdater.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.XPath;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters
+{
+    public class ConnectionStringsConfigUpdater : IUpdater<ConfigFile>
```
Comment (twsouthwick) [inline src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/ConnectionStringsConfigUpdater.cs:19]: This looks like a lot of duplicated code from the appsetting one. Can you create a shared base class or something? Or combine it into that one?

**Review by twsouthwick — CHANGES_REQUESTED** [review]

File: src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/ConnectionStringsConfigUpdater.cs (hunk 1 above)
Comment (johanbenschop) [reply src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/ConnectionStringsConfigUpdater.cs]: I've moved the duplicated code into a shared base class. Let me know what you think of this.

Comment (twsouthwick) [reply src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/ConnectionStringsConfigUpdater.cs]: Great, thanks!

**Review by twsouthwick — APPROVED** [review]
Thanks @johanbenschop for the contribution. LGTM!

**Pull request merged**